### PR TITLE
Doc: Add a single table as summary to math documentation

### PR DESCRIPTION
**Pull request by Nodd**
This is an draft for a table summary for the math module, as discussed in [discourse](https://discuss.python.org/t/summary-tables-as-an-api-overview/68474/1). This is the version with a single table, with intermediate headers.

[Direct link to the page preview](https://cpython-previews--125810.org.readthedocs.build/en/125810/library/math.html)

There is an alternative with [multiple tables](https://github.com/python/cpython/pull/125809)

----
📚 Documentation preview 📚: https://cpython-previews--125810.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/math.rst [hunk 1]
```diff
@@ -26,6 +26,92 @@ The following functions are provided by this module.  Except when explicitly
...
+:func:`gamma`            `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_.
+:func:`lgamma`           Natural logarithm of the absolute value of the `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_.
+
+**Constants**
+----------------------------------------------------------------------------------------
+:data:`pi`               *π* = 3.141592...
+:func:`e`                *e* = 2.718281...
+:func:`tau`              *τ* = 2\ *π* = 6.283185....
+:func:`inf`              Positive infinity.
+:func:`nan`              "Not a number" (NaN).
```
Comment (merwok) [inline Doc/library/math.rst]: These are data, not func — func adds parens!

Comment (Nodd) [reply Doc/library/math.rst]: Indeed, I forgot to fix after the previous line. Thanks for noticing, I fixed it.

**Comment by nedbat** [comment]
Thanks for doing this, and linking directly to the preview.

I think the summary would be a little better if the arguments were shown, especially since sometimes a value mentioned in the summary is an argument, and sometimes a constant:

- exp(x): e raised to the power of x
- pow(x, y): x raised to the power of y

**Comment by Nodd** [comment]
I thought about that too, and I even started to implement it. It looks indeed better, but it has to be done manually, so it's a bit tedious. If someone knows a better way than this:
```rst
:func:`pow(x, y) <pow>`
```
I'm all ears!

**Comment by merwok** [comment]
The parens are added by a sphinx setting (wrongly, I always thought, as `func()` is not the same as `func` or `func(a, b)` – with marked-up doc we don’t need the parens to signify that this is a function), so maybe there is another setting to show signatures? Then again maybe not, as signatures can be quite heavy with annotations…

**Comment by Nodd** [comment]
I added the arguments. I choose to not mark default, positional-only nor keyword-only arguments in the table to keep it as light as possible.

**Comment by dg-pb** [comment]
👍 

I like single table much more.
For me, multiple tables take more time to focus and find what I need.

FYI, There is an open PR for string methods: https://github.com/python/cpython/pull/1709

File: Doc/library/math.rst [hunk 2]
```diff
@@ -26,6 +26,92 @@ The following functions are provided by this module.  Except when explicitly
...
+:func:`gamma(x) <gamma>`                              `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_ at *x*.
+:func:`lgamma(x) <lgamma>`                            Natural logarithm of the absolute value of the `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_ at *x*.
+
+**Constants**
+--------------------------------------------------------------------------------------------------
+:data:`pi`                                            *π* = 3.141592...
+:data:`e`                                             *e* = 2.718281...
+:data:`tau`                                           *τ* = 2\ *π* = 6.283185....
+:data:`inf`                                           Positive infinity.
+:data:`nan`                                           "Not a number" (NaN).
```
Comment (hugovk) [inline Doc/library/math.rst]: I suggest removing the full stops:
```suggestion
:func:`ceil(x) <ceil>`                                Ceiling of *x*, the smallest integer greater than or equal to *x*
:func:`comb(n, k) <comb>`                             Number of ways to choose *k* items from *n* items without repetition and without order
:func:`copysign(x, y) <copysign>`                     Magnitude (absolute value) of *x* with the sign of *y*
:func:`fabs(x) <fabs>`                                Absolute value of *x*
:func:`factorial(n) <factorial>`                      *n* factorial
:func:`floor (x)  <floor>`                            Floor of *x*, the largest integer less than or equal to *x*
:func:`fma(x, y, z) <fma>`                            Fused multiply-add operation: ``(x * y) + z``
:func:`fmod(x, y) <fmod>`                             Remainder of division ``x / y``
:func:`frexp(x) <frexp>`                              Mantissa and exponent of *x*
:func:`fsum(iterable) <fsum>`                         Sum of values in the input *iterable*
:func:`gcd(*integers) <gcd>`                          Greatest common divisor of the integer arguments
:func:`isclose(a, b, rel_tol, abs_tol) <isclose>`     Check if the values *a* and *b* are close to each other
:func:`isfinite(x) <isfinite>`                        Check if *x* is neither an infinity nor a NaN
:func:`isinf(x) <isinf>`                              Check if *x* is a positive or negative infinity
:func:`isnan(x) <isnan>`                              Check if *x* is a NaN  (not a number)
:func:`isqrt(n) <isqrt>`                              Integer square root of a nonnegative integer *n*
:func:`lcm(*integers) <lcm>`                          Least common multiple of the integer arguments
:func:`ldexp(x, i) <ldexp>`                           ``x * (2**i)``, inverse of function :func:`frexp`
:func:`modf(x) <modf>`                                Fractional and integer parts of *x*
:func:`nextafter(x, y, steps) <nextafter>`            Floating-point value *steps* steps after *x* towards *y*
:func:`perm(n, k) <perm>`                             Number of ways to choose *k* items from *n* items without repetition and with order
:func:`prod(iterable, start) <prod>`                  Product of elements in the input *iterable* with a *start* value
:func:`remainder(x, y) <remainder>`                   Remainder of *x* with respect to *y*
:func:`sumprod(p, q) <sumprod>`                       Sum of products from two iterables *p* and *q*
:func:`trunc(x) <trunc>`                              Integer part of *x*
:func:`ulp(x) <ulp>`                                  Value of the least significant bit of *x*

**Power and logarithmic functions**
--------------------------------------------------------------------------------------------------
:func:`cbrt(x) <cbrt>`                                Cube root of *x*
:func:`exp(x) <exp>`                                  *e* raised to the power *x*
:func:`exp2(x) <exp2>`                                *2* raised to the power *x*
:func:`expm1(x) <expm1>`                              *e* raised to the power *x*, minus 1
:func:`log(x, base) <log>`                            Logarithm of *x* to the given base (*e* by default)
:func:`log1p(x) <log1p>`                              Natural logarithm of *1+x* (base *e*)
:func:`log2(x) <log2>`                                Base-2 logarithm of *x*
:func:`log10(x) <log10>`                              Base-10 logarithm of *x*
:func:`pow(x, y) <pow>`                               *x* raised to the power *y*
:func:`sqrt(x) <sqrt>`                                Square root of *x*

**Trigonometric functions**
--------------------------------------------------------------------------------------------------
:func:`acos(x) <acos>`                                Arc cosine of *x*
:func:`asin(x) <asin>`                                Arc sine of *x*
:func:`atan(x) <atan>`                                Arc tangent of *x*
:func:`atan2(y, x) <atan2>`                           ``atan(y / x)``
:func:`cos(x) <cos>`                                  Cosine of *x*
:func:`dist(p, q) <dist>`                             Euclidean distance between two points *p* and *q* given as an iterable of coordinates
:func:`hypot(*coordinates) <hypot>`                   Euclidean norm of an iterable of coordinates
:func:`sin(x) <sin>`                                  Sine of *x*
:func:`tan(x) <tan>`                                  Tangent of *x*

**Angular conversion**
--------------------------------------------------------------------------------------------------
:func:`degrees(x) <degrees>`                          Convert angle *x* from radians to degrees
:func:`radians(x) <radians>`                          Convert angle *x* from degrees to radians

**Hyperbolic functions**
--------------------------------------------------------------------------------------------------
:func:`acosh(x) <acosh>`                              Inverse hyperbolic cosine of *x*
:func:`asinh(x) <asinh>`                              Inverse hyperbolic sine of *x*
:func:`atanh(x) <atanh>`                              Inverse hyperbolic tangent of *x*
:func:`cosh(x) <cosh>`                                Hyperbolic cosine of *x*
:func:`sinh(x) <sinh>`                                Hyperbolic sine of *x*
:func:`tanh(x) <tanh>`                                Hyperbolic tangent of *x*

**Special functions**
--------------------------------------------------------------------------------------------------
:func:`erf(x) <erf>`                                  `Error function <https://en.wikipedia.org/wiki/Error_function>`_ at *x*
:func:`erfc(x) <erfc>`                                `Complementary error function <https://en.wikipedia.org/wiki/Error_function>`_ at *x*
:func:`gamma(x) <gamma>`                              `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_ at *x*
:func:`lgamma(x) <lgamma>`                            Natural logarithm of the absolute value of the `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_ at *x*

**Constants**
--------------------------------------------------------------------------------------------------
:data:`pi`                                            *π* = 3.141592...
:data:`e`                                             *e* = 2.718281...
:data:`tau`                                           *τ* = 2\ *π* = 6.283185...
:data:`inf`                                           Positive infinity
:data:`nan`                                           "Not a number" (NaN)
```

Comment (Nodd) [reply Doc/library/math.rst]: I took the formatting from [`statistics`](https://docs.python.org/3/library/statistics.html), but I agree that it looks better without them, and they are not present in other tables like [curses.ascii](https://docs.python.org/3/library/curses.ascii.html).

File: Doc/library/math.rst [hunk 3]
```diff
@@ -26,6 +26,92 @@ The following functions are provided by this module.  Except when explicitly
 noted otherwise, all return values are floats.
 
 
+====================================================  ============================================
+**Number-theoretic and representation functions**
```
Comment (skirpichev) [inline Doc/library/math.rst:30]: I think this should be split out.

For some inspiration you could take look on the C17 and sections in the Annex F.10, e.g. here: https://en.cppreference.com/w/c/numeric/math.  So, we should have something like 1) "Floating-point manipulation functions", 2) "Classification and comparison", 3) "Nearest integer operations", 4) "Basic operations" (maybe prod/fsum/etc - should be here).

Number-theoretic/combinatorial functions deserve a separate section.

File: Doc/library/math.rst [hunk 4]
```diff
@@ -26,6 +26,92 @@ The following functions are provided by this module.  Except when explicitly
...
+
+**Trigonometric functions**
+--------------------------------------------------------------------------------------------------
+:func:`acos(x) <acos>`                                Arc cosine of *x*
+:func:`asin(x) <asin>`                                Arc sine of *x*
+:func:`atan(x) <atan>`                                Arc tangent of *x*
+:func:`atan2(y, x) <atan2>`                           ``atan(y / x)``
+:func:`cos(x) <cos>`                                  Cosine of *x*
+:func:`dist(p, q) <dist>`                             Euclidean distance between two points *p* and *q* given as an iterable of coordinates
+:func:`hypot(*coordinates) <hypot>`                   Euclidean norm of an iterable of coordinates
```
Comment (skirpichev) [inline Doc/library/math.rst:80]: I think this either belongs to the above section (powers), or to a separate section, perhaps with fsum, prod, sumprod.

File: Doc/library/math.rst [hunk 5]
```diff
@@ -26,6 +26,92 @@ The following functions are provided by this module.  Except when explicitly
...
+--------------------------------------------------------------------------------------------------
+:func:`cbrt(x) <cbrt>`                                Cube root of *x*
+:func:`exp(x) <exp>`                                  *e* raised to the power *x*
+:func:`exp2(x) <exp2>`                                *2* raised to the power *x*
+:func:`expm1(x) <expm1>`                              *e* raised to the power *x*, minus 1
+:func:`log(x, base) <log>`                            Logarithm of *x* to the given base (*e* by default)
+:func:`log1p(x) <log1p>`                              Natural logarithm of *1+x* (base *e*)
+:func:`log2(x) <log2>`                                Base-2 logarithm of *x*
+:func:`log10(x) <log10>`                              Base-10 logarithm of *x*
+:func:`pow(x, y) <pow>`                               *x* raised to the power *y*
```
Comment (skirpichev) [inline Doc/library/math.rst]: This should point to the math.pow, not to the builtin.

File: Doc/library/math.rst (hunk 3 above)
Comment (Nodd) [reply Doc/library/math.rst:30]: Thank you for your suggestion. I simply used the current organization of the page, and copied the existing section labels in the page. I think that reorganizing the whole page can be made in another PR, to focus this one on adding the table. I can do it afterwards.

File: Doc/library/math.rst (hunk 4 above)
Comment (Nodd) [reply Doc/library/math.rst:80]: Same as above, it can be modified in a following PR.

File: Doc/library/math.rst (hunk 5 above)
Comment (Nodd) [reply Doc/library/math.rst]: That's a very good catch! I'll fix this ASAP.

**Review by willingc — APPROVED** [review]
This looks like a nice improvement to the docs as written. My recommendation would be to merge. If necessary, we can iterate on future PRs.

Thanks @Nodd for your first contribution :tada:

As an FYI, you can ask for folks to rereview here instead of discuss.python.org.

**Comment by Nodd** [comment]
Thanks @willingc, I wasn't sure where was the best to ask for review. I'll keep it in mind for next time.

**Review by picnixz — APPROVED** [review]
I saw that you were just re-using the same titles as the sections as well as their order so I think it's ok *for this PR*. However, in a follow-up PR we should address the following:

- "Angular conversion" should be put before trigonometric functions so that trigonometric and hyperbolic ones are next to each other.
- We should say "Power, **exponential** and logarithmic functions" and not just "Power and logarithmic functions" because `exp(x)` is not a power function per se.

**Comment by skirpichev** [comment]
@picnixz, see also https://github.com/python/cpython/pull/125810#discussion_r1814366833

**Comment by willingc** [comment]
Folks, I'm merging this as is. Future PRs can iterate on sections and wording. Thanks!

**Pull request merged**